### PR TITLE
docs: improve `closure-action` rule examples

### PR DESCRIPTION
**Pull request by Caltor**
It wasn't clear before which code snippets related to what. I have split them up with a decent markdown header for each so it is clear what is going on. It took me several read-throughs before to understand these examples.

### CODE REVIEW

File: docs/rules/closure-actions.md [hunk 1]
```diff
@@ -13,12 +13,14 @@ export default Controller.extend({
 });
 ```
 
+## GOOD
```
Comment (bmish) [inline docs/rules/closure-actions.md]: Can you switch to `###` for the subheaders? Only the main header at the top is supposed to use `##`.

**Comment by bmish** [comment]
Can you squash into one commit?

**Comment by Caltor** [comment]
> Can you squash into one commit?
Can I do that through the GitHub web interface?

**Comment by bmish** [comment]
I usually do it with:
```
git rebase --interactive master
<fixup/squash into a single commit>
git push --force
```

**Comment by Caltor** [comment]
> Can you squash into one commit?

Done!

**Pull request merged**